packages/local-environment/src/nomadenv.ts

```diff
@@ -212,76 +238,91 @@ export class NomadEnv {
     };
   }
 
-  
-
   //Input arguments to d.up to disable a specific agent.
-  async up() {
-    let metrics = 9000;
+  async up(): Promise<void> {
+    const metrics = 9000;
     await Promise.all(this.domains.map((d, i) => d.up(metrics + i * 10)));
   }
 
-  async down() {
+  async down(): Promise<void> {
     await Promise.all(this.domains.map((d) => d.down()));
   }
 
   //Input arguments to d.up to disable a specific agent.
-  async upAgents() {
-    let metrics = 9000;
+  async upAgents(): Promise<void> {
+    const metrics = 9000;
     await Promise.all(this.domains.map((d, i) => d.upAgents(metrics + i * 10)));
   }
 
-  async downAgents() {
+  async downAgents(): Promise<void> {
     await Promise.all(this.domains.map((d) => d.down()));
   }
 
-  async upNetworks() {
+  async upNetworks(): Promise<void> {
     // Await domains to up networks.
     await Promise.all(this.domains.map((d) => d.networkUp()));
   }
 }
 
-export async function defaultStart() {
+export async function defaultStart(): Promise<NomadEnv> {
   // Ups 2 new hardhat test networks tom and jerry to represent home chain and target chain.
   const log = bunyan.createLogger({ name: "localenv" });
 
-  // Instantiate HardhatNetworks
-  const t = new HardhatNetwork("tom", 1);
-  const j = new HardhatNetwork("jerry", 2);
+  let tDomainNumber = 1;
+  let jDomainNumber = 2;
 
-  // Instantiate Nomad domains
-  const tDomain = new NomadDomain(t);
-  const jDomain = new NomadDomain(j);
-
-  // Await domains to up networks.
-  await Promise.all([tDomain.network.up(), jDomain.network.up()]);
+  if (process.env.tDomainNumber) {
+    tDomainNumber = parseInt(process.env.tDomainNumber);
+  }
 
-  log.info(`Upped Tom and Jerry`);
+  if (process.env.jDomainNumber) {
+    jDomainNumber = parseInt(process.env.jDomainNumber);
+  }
```

Comment:
This. Why do we need to fetch domain from somewhere else? In this case we would maybe also want to fetch domain's name from somewhere else too? (the last question is not a call to action, but rather a call to thinking, but first question might be a call to discussion)

Reply:
^ See above, not sure if you think removing this is optimal (in consideration for network forking to replicate the real chain ID

Reply:
it is not a chainId, it is domain number, which is only according to Nomad/or user. It has nothing to do with the chainId

Reply:
Removed